File: amazia-core/src/main/java/com/example/user/controller/UpdateUserController.java
package com.example.user.controller;

import com.example.user.dto.UpdateUserRequest;
import com.example.user.dto.UserResponse;
import com.example.user.service.UpdateUserService;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/users")
public class UpdateUserController {

    private final UpdateUserService updateUserService;

    public UpdateUserController(UpdateUserService updateUserService) {
        this.updateUserService = updateUserService;
    }

    @PutMapping("/{id}")
    public ResponseEntity<UserResponse> update(@PathVariable Long id,
                                               @Valid @RequestBody UpdateUserRequest request) {
        return ResponseEntity.ok(updateUserService.update(id, request));
    }
}

File: amazia-console/app/User/Controller/UpdateUserController.php
<?php

namespace App\User\Controller;

use App\Http\Controllers\Controller;
use App\User\Service\UpdateUserService;
use Illuminate\Http\Request;

class UpdateUserController extends Controller
{
    private UpdateUserService $service;

    public function __construct(UpdateUserService $service)
    {
        $this->service = $service;
    }

    public function __invoke(Request $request, int $id)
    {
        $response = $this->service->update($id, $request->all());
        return response()->json($response->json(), $response->status());
    }
}
